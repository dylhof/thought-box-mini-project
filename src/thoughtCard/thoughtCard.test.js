import React from 'react';
import { shallow } from 'enzyme';

import { ThoughtCard } from './ThoughtCard.js';

const mockThought = { title: 'Hello', body: 'World', id: 0 };
const deleteThoughtMock = jest.fn();

describe('ThoughtCard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <ThoughtCard 
        thought={mockThought}
        deleteThought={deleteThoughtMock}
      />
    )
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call deleteThought when remove button is clicked', () => {
    wrapper.find('.card-delete-btn').simulate('click');
    expect(deleteThoughtMock).toBeCalledWith(0)
  })

});
