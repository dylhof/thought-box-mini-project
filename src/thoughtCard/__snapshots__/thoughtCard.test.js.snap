// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThoughtCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThoughtCard
    deleteThought={[MockFunction]}
    thought={
      Object {
        "body": "World",
        "id": 0,
        "title": "Hello",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Hello
        </h3>,
        <p>
          World
        </p>,
        <button
          className="card-delete-btn"
          onClick={[Function]}
        >
          Remove Thought
        </button>,
      ],
      "className": "card-div",
      "id": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello",
        },
        "ref": null,
        "rendered": "Hello",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "World",
        },
        "ref": null,
        "rendered": "World",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove Thought",
          "className": "card-delete-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Thought",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Hello
          </h3>,
          <p>
            World
          </p>,
          <button
            className="card-delete-btn"
            onClick={[Function]}
          >
            Remove Thought
          </button>,
        ],
        "className": "card-div",
        "id": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello",
          },
          "ref": null,
          "rendered": "Hello",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "World",
          },
          "ref": null,
          "rendered": "World",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove Thought",
            "className": "card-delete-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Thought",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
